website/src/models/group.js

```diff
@@ -89,22 +89,15 @@ GroupSchema.pre('remove', function(next) {
   var group = this;
   async.waterfall([
     function(cb) {
-      var invitationQuery = {};
-      var groupType = group.type;
-      //Add an 's' to group type guild because the model has the plural version
-      if (group.type == "guild") groupType += "s";
-      invitationQuery['invitations.' + groupType + '.id'] = group._id;
```

Comment:
I ran a test query against invitations.party.id, and it took >60s to run. However, we still experienced timeouts after reverting this PR, so there's still more to be found.